force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -124,18 +143,26 @@ public with sharing class ClassesObjectsMethods {
     public static String findTheScientist() {
         // The list of scientists' names
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
-
         // The variable to store the concatenated string
-
+        String names;
         // Loop through the list of scientists' names
-
+        for(Integer i = 0; i < scientistNames.size(); i++){
             // If the name doesn't contain the letter 'G', skip this iteration
-
-            // Add the name to the result string, followed by a comma
-
+            if(scientistNames[i].containsIgnoreCase('g') == true){
```

Comment:
```suggestion
            if(scientistNames[i].containsIgnoreCase('g')){
```
the conidtion in the loop will only be evaluated if the predicate is true. You don't need the explicit check.